smart_campus/app/views.py

```diff
@@ -113,23 +120,158 @@ def index(request):
 
 @login_required
 def station_list_page(request):
-    context = {'email': request.user.email}
+    """Show all stations managed by the user's group"""
+    if request.user.can(Permission.ADMIN):
+        stations = Station.objects.all()
+    else:
+        stations = Station.objects.filter(owner_group=request.user.group)
+
+    station_data = [
+        {
+            'id': station.id,
+            'name': station.name,
+            'image_url': station.primary_image_url,
+        }
+        for station in stations
+    ]
+
+    context = {'email': request.user.email, 'stations': station_data}
+
     return render(request, 'app/station_list.html', context)
 
 
 @login_required
-def add_station_page(request):
-    context = {'email': request.user.email, 'categories': StationCategory.objects.all()}
-    return render(request, 'app/add_station.html', context)
+def station_edit_page(request, pk):
+    station = get_object_or_404(Station, pk=pk)
+    if request.method == 'POST':
+        # Use custom StationForm to validate form datas
+        # is_valid() will be true if received data is good
+        # There'll be error messages in 'form' instance if validation failed
+        # cleaned_data will be the form inputs from the request
+        form = StationForm(request.POST, request.FILES, instance=station)
+        if form.is_valid():
+            data = form.cleaned_data
+            station = form.save(commit=False)
+            station.location = Point(x=data['lng'], y=data['lat'])
+
+            # Clear linked Beacons
+            station.beacon_set.clear()
+            station.beacon_set.add(Beacon.objects.get(name=data['beacon']))
+            station.save()
+
+            # Handle change of images
+            if request.POST.get('img_changed') == 'true':
+                # Clear old images
+                for img in StationImage.objects.filter(station=station):
+                    # Clear files on the disk
+                    # Delete model instance won't delete them
+                    if os.path.isfile(img.image.path):
+                        os.remove(img.image.path)
+                    img.delete()
+
+                # Add new images
+                for img_num in range(1, settings.MAX_IMGS_UPLOAD + 1):
+                    img_name = 'img{0}'.format(img_num)
+                    if data[img_name]:
+                        StationImage.objects.create(
+                            station=station,
+                            image=data[img_name],
+                            is_primary=True if img_num == data['main_img_num'] else False
```

Comment:
Write a statement here is kind of complicated.
I suggest you can save it as a variable.
e.g. 
```python
is_primary = ...
StationImage... (
    is_primary=is_primary
)
```